app/goal_routes.py

```
@@ -0,0 +1,96 @@
+import datetime, os, requests
+from .routes import validate_model
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from flask import Blueprint, jsonify, make_response, request, abort
+from dotenv import load_dotenv
+
+load_dotenv()
+
+goals_bp = Blueprint('goals', __name__, url_prefix='/goals')
+
+@goals_bp.route("", methods=["GET"])
+def read_all_goals():
+    title_query = request.args.get('title')
+
+    if title_query:
+        goals = Goal.query.filter_by(title=title_query)
+        
+    if not title_query:
+        goals = Goal.query.all()
+    goals_response = [goal.goal_to_dict() for goal in goals]
+    return jsonify(goals_response)
+
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    if not request_body:
+        return {"details": "Invalid data"}, 400
```

This will handle a client's request if the request body is empty (falsey), but if a client sent over a request body that didn't have the required field "title" then line 31 will throw a KeyError and the client's response will be a 500 error instead of your nice error response. For example: request body was {"name": "declutter the house"}

You could add some error handling logic by incorporating a try/except block so if a KeyError does arise then you can return an error message.